### PR TITLE
ROX-12814: Disable PolicyFieldsTest on openshift.

### PR DESCRIPTION
**Pull request by porridge**
## Description

Scanner OOMs pretty reliably on this test on osd jobs for over 2 weeks.
Disabling for now according to [our policy](https://srox.slack.com/archives/CELUQKESC/p1667906225265369).

## Checklist
- [ ] Investigated and inspected CI test results to make sure this test was skipped
- [x] ~Unit test and regression tests added~
- [x] ~Evaluated and added CHANGELOG entry if required~
- [x] ~Determined and documented upgrade steps~
- [x] ~Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

## Testing Performed

Relying on CI.
/test all

### CODE REVIEW

File: qa-tests-backend/src/test/groovy/PolicyFieldsTest.groovy [hunk 1]
```diff
@@ -18,10 +18,13 @@ import org.junit.Assume
 import org.junit.experimental.categories.Category
 import services.AlertService
 import services.PolicyService
+import spock.lang.IgnoreIf
 import spock.lang.Shared
 import spock.lang.Unroll
 import util.Env
 
+// TODO(ROX-12814): Scanner OOMs on this test in some Openshift jobs.
+@IgnoreIf({ Env.mustGetOrchestratorType() == OrchestratorTypes.OPENSHIFT })
```
Comment (gavin-stackrox) [inline qa-tests-backend/src/test/groovy/PolicyFieldsTest.groovy:27]: What about only disabling for OSD GCP? Other openshifts do not have this OOM IIRC?
```suggestion
@IgnoreIf({ Env.CI_JOBNAME.contains("osd-gcp") })
```

**Review by gavin-stackrox — CHANGES_REQUESTED** [review]

File: qa-tests-backend/src/test/groovy/PolicyFieldsTest.groovy (hunk 1 above)
Comment (porridge) [reply qa-tests-backend/src/test/groovy/PolicyFieldsTest.groovy:27]: There are mentions of failures on `rosa` and `openshift-oldest` too in the ticket :-/

**Review by gavin-stackrox — APPROVED** [review]
Sorry! I missed those in the hidden comments!

**Review by janisz — APPROVED** [review]

**Comment by porridge** [comment]
Looks like the set of tests that run on per-submit is smaller than the set run on nightlies.
While this test is explicitly mentioned [in this nightly `openshift-newest-qa` run](https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/branch-ci-stackrox-stackrox-nightlies-openshift-newest-qa-e2e-tests/1592022230034812928) for example, it is not mentioned at all in the corresponding run for this PR (rather than mentioned as skipped).
So I'm going to merge this, since checking tomorrow's nightly is probably the least effort way to check that this change worked as intended.

**Pull request merged**